contracts/staking/test/unit_tests/mixin_vault_core_test.ts

```diff
@@ -0,0 +1,99 @@
+import { blockchainTests, expect, filterLogsToArguments } from '@0x/contracts-test-utils';
+import { StakingRevertErrors } from '@0x/order-utils';
+import { OwnableRevertErrors } from '@0x/utils';
+
+import { constants } from '../utils/constants';
+
+import {
+    artifacts,
+    TestMixinVaultCoreContract,
+    TestMixinVaultCoreInCatastrophicFailureModeEventArgs,
+    TestMixinVaultCoreStakingContractChangedEventArgs,
+} from '../../src';
+
+blockchainTests.resets('MixinVaultCore', env => {
+    let owner: string;
+    let nonOwnerAddresses: string[];
+    let testContract: TestMixinVaultCoreContract;
+
+    before(async () => {
+        [owner, ...nonOwnerAddresses] = await env.getAccountAddressesAsync();
+
+        testContract = await TestMixinVaultCoreContract.deployFrom0xArtifactAsync(
+            artifacts.TestMixinVaultCore,
+            env.provider,
+            env.txDefaults,
+            artifacts,
+        );
+    });
+
+    describe('Set staking contract', () => {
```

Comment:
I like the thoroughness of these tests, but I think we should also have test suites that only unit test the modifiers. These test the modifiers in a way that is a little indirect for my taste.

This said, this is just my opinion. I'd be interested to hear whether you think this is reasonable or not.

Reply:
hm yeah it could be cool to have some common testing framework specifically for modifiers, e.g. to test nesting behavior